sqlx-macros/src/query/mod.rs

```diff
@@ -16,19 +17,31 @@ use crate::database::DatabaseExt;
 use crate::query::data::QueryData;
 use crate::query::input::RecordType;
 use either::Either;
+use lazy_static::lazy_static;
 
 mod args;
 mod data;
 mod input;
 mod output;
 
-pub fn expand_input(input: QueryMacroInput) -> crate::Result<TokenStream> {
-    let manifest_dir =
-        env::var("CARGO_MANIFEST_DIR").map_err(|_| "`CARGO_MANIFEST_DIR` must be set")?;
+lazy_static! {
+    static ref CRATE_ROOT: PathBuf = {
+        let manifest_dir =
+            env::var("CARGO_MANIFEST_DIR").expect("`CARGO_MANIFEST_DIR` must be set");
+
+        let metadata = cargo_metadata::MetadataCommand::new()
+            .current_dir(manifest_dir)
+            .exec()
+            .expect("Could not fetch metadata");
+
+        metadata.workspace_root
+    };
+}
 
-    // If a .env file exists at CARGO_MANIFEST_DIR, load environment variables from this,
+pub fn expand_input(input: QueryMacroInput) -> crate::Result<TokenStream> {
+    // If a .env file exists at CRATE_ROOT, load environment variables from this,
     // otherwise fallback to default dotenv behaviour.
-    let env_path = std::path::Path::new(&manifest_dir).join(".env");
+    let env_path = CRATE_ROOT.join(".env");
```

Comment:
The current behavior is to use `dotenv::from_path` iff the resolved `env_path` exists. 

This won't be the case for workspaces because `CARGO_MANIFEST_DIR` is the crate root and not the workspace root. 

Are we ok to fallback to `dotenv::dotenv()` in the case that `env_path` does not exist? Should that behavior be behind a feature flag?